unstructured/ingest/connector/sftp.py

```diff
@@ -0,0 +1,56 @@
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Type
+from urllib.parse import urlparse
+
+from unstructured.ingest.connector.fsspec import (
+    FsspecIngestDoc,
+    FsspecSourceConnector,
+    SimpleFsspecConfig,
+)
+from unstructured.ingest.error import SourceConnectionError
+from unstructured.utils import requires_dependencies
+
+
+@dataclass
+class SimpleSftpConfig(SimpleFsspecConfig):
+    host: str = ""
+    port: str = ""
+
+    def __post_init__(self):
+        super().__post_init__()
+
+        _, ext = os.path.splitext(self.remote_url)
+        if ext:
+            # We only want the file_path if it has an extension
+            self.file_path = Path(self.remote_url).name
+            self.dir_path = Path(urlparse(self.remote_url).path).parent.as_posix().lstrip("/")
```

Comment:
small nit, since we're using `urlparse(self.remote_url)` so often, can we create a variable and use that everywhere instead? 
```python
parsed_url = urlparse(self.remote_url)
```

Reply:
Definitely.